### PR TITLE
crowdstrike: Modify threat.framework based on prefix and custom tactic names

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
```
Derive threat.framework from prefix and custom tactics.

Crowdstrike adds custom tactics based on Falcon Detection framework.
The ID of tactics starts with prefix `CS`. In cases where this is 
missing, all the custom Crowdstrike tactics names are used to set 
`threat.framework` as `CrowdStrike Falcon Detections Framework`.
```

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

`elastic-package test pipeline --generate -v --data-streams=alert`
```
--- Test results for package: crowdstrike - START ---
╭─────────────┬─────────────┬───────────┬────────────────┬────────┬──────────────╮
│ PACKAGE     │ DATA STREAM │ TEST TYPE │ TEST NAME      │ RESULT │ TIME ELAPSED │
├─────────────┼─────────────┼───────────┼────────────────┼────────┼──────────────┤
│ crowdstrike │ alert       │ pipeline  │ test-alert.log │ PASS   │   3.428458ms │
╰─────────────┴─────────────┴───────────┴────────────────┴────────┴──────────────╯
--- Test results for package: crowdstrike - END   ---
Done
```

## Related issues

- Closes https://github.com/elastic/integrations/issues/10196

### CODE REVIEW

File: packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml [hunk 1]
```diff
@@ -232,6 +232,19 @@ processors:
...
+      description: First tactic from crowdstrike.event.TacticIds is used to determine threat.framework
+      tag: set_threat_framework_cs
+      if: ctx.crowdstrike?.event?.TacticIds != null && ctx.crowdstrike.event.TacticIds.startsWith("CS")
+  - set: 
+      field: threat.framework
+      value: MITRE ATT&CK
+      description: First tactic from crowdstrike.event.TacticIds is used to determine threat.framework
+      override: false
+      tag: set_threat_framework
+      if: ctx.threat?.framework == null && ctx.crowdstrike?.event?.TacticIds != null && ctx.crowdstrike.event.TacticIds.startsWith("TA")
```
Comment (kcreddy) [inline packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml:247]: For Falcon XDR data, both tactics and techniques are a comma-separated values. The first tacticId is chosen to derive threat.framework

File: packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml [hunk 2]
```diff
@@ -232,6 +229,21 @@ processors:
...
       tag: split_tactic_ids
       if: ctx.crowdstrike?.event?.TacticIds != null
+  - set: 
+      field: threat.framework
+      value: CrowdStrike Falcon Detections Framework
+      description: For Crowdstrike Falcon Detection tactics and/or that are prefixed with "CS". First tactic from crowdstrike.event.TacticIds is used to determine threat.framework
+      tag: set_threat_framework_cs
+      if: >-
+        (ctx.crowdstrike?.event?.TacticIds != null && ctx.crowdstrike.event.TacticIds.startsWith("CS"))
+        || (ctx.crowdstrike?.event?.TechniqueIds != null && ctx.crowdstrike.event.TechniqueIds.startsWith("CS"))
```
Comment (kcreddy) [inline packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml:239]: For Falcon XDR data, both tactics and techniques are a comma-separated values. The first tactic values are chosen to derive `threat.framework`

**Review by chemamartinez — APPROVED** [review]
LGTM

File: packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml (hunk 2 above)
Comment (efd6) [reply packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml:239]: Do we ever see cases where it is "CSTxxxxx,Tyyyyy" (i.e. mixed)?

Comment (kcreddy) [reply packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml:239]: Actually, I only have the sample logs as the data points, in which we dont see such mixed cases.

Comment (efd6) [reply packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml:239]: So should we check all of them and add to deduped frameworks?

Comment (kcreddy) [reply packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml:239]: I also thought of doing that, but unlike `threat.tactic.*` ECS fields, [`threat.framework`](https://www.elastic.co/guide/en/ecs/current/ecs-threat.html#field-threat-framework) describes it as a singular value.

Comment (efd6) [reply packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml:239]: OK

Comment (kcreddy) [reply packages/crowdstrike/data_stream/falcon/elasticsearch/ingest_pipeline/xdr_detection_summary.yml]: Updated in [6e862ce](https://github.com/elastic/integrations/pull/10256/commits/6e862ce8ce08a656632f29a648e7b703675c9cb4)

**Review by efd6 — APPROVED** [review]

**Pull request merged**